python/pyarrow/_csv.pyx

```diff
@@ -223,6 +296,34 @@ cdef class ReadOptions(_Weakrefable):
         - `skip_rows` is applied (if non-zero);
         - column names aread (unless `column_names` is set);
         - `skip_rows_after_names` is applied (if non-zero).
+
+        Examples:
+        ---------
+
+        >>> from pyarrow import csv
+
+        >>> read_options = csv.ReadOptions(skip_rows_after_names=1)
+        >>> csv.read_csv("animals.csv", read_options=read_options)
+        pyarrow.Table
+        animals: string
+        n_legs: int64
+        entry: string
+        ----
+        animals: [["Horse","Brittle stars","Centipede"]]
+        n_legs: [[4,5,100]]
+        entry: [["02/03/2022","03/03/2022","04/03/2022"]]
```

Comment:
Sidenote: if the dates where in ISO format (e.g. "2022-03-02"), they would be inferred neatly as date32.

Reply:
Yes, true, it works great by default like that. But I needed an example for `timestamp_parsers` =)